### ISSUE TITLE
PCC drop in SSD300VGG16 , PCC=0.13025588869684107

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#3570]
### Describe the bug

- `ssd300_vgg16` fails with `PCC=0.13025588869684107` - [mar3_ssd300_vgg16.log](https://github.com/user-attachments/files/25736098/mar3_ssd300_vgg16_1.log)
- The PCC drop originates from the chain of [softmax](https://github.com/pytorch/vision/blob/d801a34632023859a0a274803d6abaf0a45d77a5/torchvision/models/detection/ssd.py#L418C23-L418C32) -> [slice](https://github.com/pytorch/vision/blob/d801a34632023859a0a274803d6abaf0a45d77a5/torchvision/models/detection/ssd.py#L433C25-L433C41) -> [greater than](https://github.com/pytorch/vision/blob/d801a34632023859a0a274803d6abaf0a45d77a5/torchvision/models/detection/ssd.py#L435C35-L435C36) ops. `PCC=0.9601775503884877` - [mar3_sanity.log](https://github.com/user-attachments/files/25736112/mar3_sanity.log)
- ttnn.softmax introduces small numerical errors in probability values. When these values straddle the > 0.01 score threshold, the boolean comparison amplifies the tiny float differences into large boolean mismatches, causing the PCC drop.
- The sanity cannot be reduced further - all three ops (softmax + slice + gt) are required to reproduce the drop. The experiments below confirm this.
- **Drop progression in the real model** : Returned intermediate outputs to show how PCC degrades through the op chain - [mar3_ssd300_vgg16_model_run.log](https://github.com/user-attachments/files/25736216/mar3_ssd300_vgg16_model_run.log), [ssd.py](https://github.com/user-attachments/files/25736217/ssd.py)

| Output      | Description                          | PCC     |
|-------------|--------------------------------------|---------|
| cls_logits  | Input to softmax                    | 0.99989 |
| pred_scores | Softmax output (batched)             | 0.99992 |
| scores      | Softmax output (unbatched)           | 0.99992 |
| score       | Sliced column                        | 1.0     |
| keep_idxs   | Boolean after > 0.01                 | 0.87996 |

- All float-valued intermediates have PCC > 0.99. Only the final boolean output drops, confirming the threshold comparison amplifies softmax error.
- **Isolation experiments** : Each experiment uses saved inputs from the real model run (not random data) - [mar3_experiments.log](https://github.com/user-attachments/files/25736276/mar3_experiments.log)

| Test     | Ops               | PCC    |
|----------|------------------|--------|
| sanity1  | softmax only     | 0.9999 |
| sanity2  | softmax + slice  | 1.0    |
| sanity3  | gt only          | 1.0    |
| sanity4  | slice + gt       | 0.9989 |

- No individual op or pair of ops causes the drop. All three together are required.
- **Note:** The PCC differs between the full model (0.88) and the sanity (0.96) because in the sanity, both CPU and device receive the identical saved input tensor. In the full model, the input to softmax is the device-computed backbone output, which already has minor numerical differences (PCC > 0.99 but not exactly 1.0). These input-level differences compound with the softmax error, causing a larger drop.

### Repro

```
git checkout kkannan/mar3_ssdvgg_pcc_drop
git submodule update --init --recursive
git lfs pull --include '*.pt'

# model run
pytest tests/runner/test_models.py::test_all_models_torch[ssd300_vgg16/pytorch-Default-single_device-inference] -svv 

# Model run with intermediate outputs (use attached ssd.py)
# Replace model code & rerun above command

# sanity run 
pytest tests/torch/ops/test_sanity.py -svv

# Isolation experiments
pytest tests/torch/ops/test_experiments.py -svv

# emitpy  
python examples/pytorch/codegen/python/sanity.py
```

### Logs

- [sanity.zip](https://github.com/user-attachments/files/25736437/sanity.zip)
- [mar3_sanity_emitpy_run.log](https://github.com/user-attachments/files/25736436/mar3_sanity_emitpy_run.log)
